pkg/webui/console/components/mac-settings-section/index.js

```diff
@@ -0,0 +1,194 @@
+  factoryPresetFreqTitle: 'Factory Preset Frequencies',
+  factoryPresetFreqDescription: 'List of factory-preset frequencies. Note: order is respected.',
+  freqAdd: 'Add Frequency',
+})
+
+// Inconsistent mac_settings enum values see https://github.com/TheThingsNetwork/lorawan-stack/issues/2258
+
+// 0...15
+const dataRateIndexOptions = Array.from({ length: 16 }, (_, index) => ({
+  value: index,
+  label: index,
+}))
+
+// 1...15
+const rx1DelayOptions = Array.from({ length: 15 }, (_, index) => ({
+  value: index + 1,
+  label: <Message content={m.delayValue} values={{ count: index + 1 }} />,
+}))
+
+// 0...7
+const pingSlotPeriodicityOptions = Array.from({ length: 8 }, (_, index) => {
+  const value = Math.pow(2, index)
+
+  return {
+    value: `PING_EVERY_${value}S`,
+    label: <Message content={m.pingSlotPeriodicityValue} values={{ count: value }} />,
+  }
+})
+
+// 0...7
+const dataRateOffsetOptions = Array.from({ length: 8 }, (_, index) => ({
+  value: index,
+  label: index,
+}))
+
+const MacSettingsSection = props => {
+  const { activationMode, deviceClasses, resetsFCnt: initialFCnt, initiallyCollapsed } = props
+
+  const isClassB = deviceClasses.includes(DEVICE_CLASSES.CLASS_B)
+  const isOTAA = activationMode === ACTIVATION_MODES.OTAA
+  const isABP = activationMode === ACTIVATION_MODES.ABP
+
+  const [resetsFCnt, setResetsFCnt] = React.useState(isABP && initialFCnt)
+  const handleResetsFCntChange = React.useCallback(evt => {
+    const { checked } = evt.target
+
+    setResetsFCnt(checked)
+  }, [])
+
+  return (
+    <Form.CollapseSection
+      id="mac-settings"
+      title={m.macSettings}
+      initiallyCollapsed={initiallyCollapsed}
+    >
+      {isABP && (
+        <Form.Field
+          title={m.rx1DelayTitle}
+          description={m.rx1DelayDescription}
+          name="mac_settings.rx1_delay.value"
+          component={Select}
+          options={rx1DelayOptions}
+        />
+      )}
+      {(isABP || isOTAA) && (
+        <Form.Field
+          title={m.rx1DataRateOffsetTitle}
+          name="mac_settings.rx1_data_rate_offset"
+          component={Select}
+          options={dataRateOffsetOptions}
+        />
+      )}
+      {isABP && (
+        <Form.Field
+          title={m.resetsFCnt}
+          onChange={handleResetsFCntChange}
+          warning={resetsFCnt ? m.resetWarning : undefined}
+          name="mac_settings.resets_f_cnt"
+          component={Checkbox}
+        />
+      )}
+      <Form.Field
+        title={m.rx2DataRateIndexTitle}
+        description={m.rx2DataDateIndexDescription}
+        name="mac_settings.rx2_data_rate_index.value"
+        component={Select}
+        options={dataRateIndexOptions}
+      />
+      <Form.Field
+        type="number"
+        min={100000}
+        step={100}
+        title={m.rx2FrequencyTitle}
+        description={m.rx2FrequencyDescription}
+        placeholder={m.frequencyPlaceholder}
+        name="mac_settings.rx2_frequency"
+        component={Input}
```

Comment:
Not sure what you mean

Reply:
`869,525` MHz frequency needs to be entered as `869525000` Hz. I suspect that it's more intuitive for the user to be able to enter the value as MHz. This can be done via the `encode` / `decode` props.

Reply:
Actually, even better would be to use the `<UnitInput />` component here, with a default on `MHz` and `Hz` second option.